baseplate/integration/pyramid/__init__.py

```diff
@@ -67,6 +69,12 @@ def baseplate_tween(request):
     return baseplate_tween
 
 
+class BaseplatePyramidRequest(Request, _LazyAttributesMixin):
+    def __init__(self, *args, **kwargs):
+        super(BaseplatePyramidRequest, self).__init__(*args, **kwargs)
+        _LazyAttributesMixin.__init__(self)
```

Comment:
I don't think you need to do this as long as you're calling `super()`

Reply:
I wasn't sure how exactly multiple inheritance would work here, so I tried it out:

```
>>> class Base(object):
...   def __init__(self):
...     print("Base called")
... 
>>> class Mixin(object):
...   def __init__(self):
...     print("Mixin called")
... 
>>> class Implementer(Base, Mixin):
...   def __init__(self):
...     super(Implementer, self).__init__()
...     print("Implementer called")
... 
>>> Implementer()
Base called
Implementer called
<__main__.Implementer object at 0x10ff78860>
```

The `super()` function is just going to return the first class in `__mro__`, so `Mixin` doesn't get triggered by calling super.

Reply:
I think if the base classes also call super, then both will get called. 

https://stackoverflow.com/a/16310777

Reply:
Oh I see. I think this would work for our thrift request object, but not our pyramid request object. It looks like we would make it down to webob's `BaseRequest`  which doesn't appear to call `super`:
https://github.com/Pylons/webob/blob/b5297529ef5a8dd3dd84c6cfa41afd101acfa8a2/src/webob/request.py#L93

Reply:
Makes sense in it's own weird way, multiple inheritance is kind of a mess 😅